helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroCompatibilityHelper.java

```diff
@@ -925,33 +954,25 @@ private static AvroVersion detectAvroCompilerVersion() {
 
   // We are fishing for the ```file.resource.loader.class``` property in the private field `velocityEngine` of
   // a SpecificCompiler class.
-  private static boolean checkIfVelocityEngineFileResourceLoaderClassExists(Class<?> specificCompilerClass)
-      throws IllegalStateException {
-    try {
-      // If `velocityEngine` were public, we could simplify this reflection to be
-      // ```VelocityEngine velocityEngine = (new SpecificCompiler()).velocityEngine;````
-      Field velocityEngineField = specificCompilerClass.getDeclaredField("velocityEngine");
-      velocityEngineField.setAccessible(true);
-      // SpecificCompiler's default constructor is also private.
-      Constructor constructor = specificCompilerClass.getDeclaredConstructor();
-      constructor.setAccessible(true);
-      constructor.newInstance();
-      Object velocityEngine = velocityEngineField.get(constructor.newInstance());
-
-      // Without reflection, this would be ```velocityEngine.getProperty("file.resource.loader.class");```
-      Class<?> velocityEngineClass = Class.forName("org.apache.velocity.app.VelocityEngine");
-      VersionDetectionUtil.markUsedForAvroCompiler(velocityEngineClass);
-      Method velocityEngineGetPropertyMethod = velocityEngineClass.getMethod("getProperty", String.class);
-      Object fileResourceLoaderClassProperty =
-          velocityEngineGetPropertyMethod.invoke(velocityEngine, "file.resource.loader.class");
-
-      return fileResourceLoaderClassProperty != null;
-
-      // We should never reach these exceptions
-    } catch (Exception expected) {
-      throw new IllegalStateException("unexpected exception while retrieving velocity engine config",
-          expected.fillInStackTrace());
-    }
+  private static boolean checkIfVelocityEngineFileResourceLoaderClassExists(Class<?> specificCompilerClass) throws Exception {
```

Comment:
Do we want this to throw an uncaught exception if there is any issue with this method?

Reply:
Nevermind, I see where the try-catch was moved to

Reply:
we catch it in the static initializer